temporalio/common.py

```diff
@@ -729,7 +729,7 @@ def with_additional_attributes(
         ...
 
 
-class MetricCounter(MetricCommon):
+class MetricCounter(MetricCommon, ABC):
```

Comment:
This is interesting. Is re-inheriting ABC at every level a general Python preference or just a preference of a certain tool?

Reply:
Pyright was telling us to do it since they were effectively still ABCs

https://docs.basedpyright.com/v1.31.0/configuration/config-files/#reportImplicitAbstractClass

We can disable any we don't like, but I was thinking that we should err on the side of going with them.

<img width="1219" height="302" alt="image" src="https://github.com/user-attachments/assets/e041f620-1bba-46b6-9191-dc3075168044" />